torch/onnx/_internal/exporter.py

```diff
@@ -0,0 +1,309 @@
+# necessary to surface onnx.ModelProto through ExportOutput:
+from __future__ import annotations
+
+import abc
+import io
+import logging
+from typing import (
+    Any,
+    Callable,
+    Final,
+    Mapping,
+    Optional,
+    Protocol,
+    runtime_checkable,
+    Sequence,
+    TYPE_CHECKING,
+    TypeVar,
+    Union,
+)
+
+import torch
+
+from torch.onnx._internal import _beartype
+
+# We can only import onnx from this module in a type-checking context to ensure that
+# 'import torch.onnx' continues to work without having 'onnx' installed. We fully
+# 'import onnx' inside of dynamo_export (by way of _assert_dependencies).
+if TYPE_CHECKING:
+    import onnx
+
+
+_DEFAULT_OPSET_VERSION: Final[int] = 18
```

Comment:
I wonder if we could put this into torch.onnx._constants?

Reply:
I'd like to keep it separate for now since `_constants` is for the torchscript path.